### PR TITLE
Abs looks good

### PR DESCRIPTION
**Pull request by RussNelson**
AudioBufferSource looks clean. Just improving a couple of comments. My only concern is that AudioSourceChannelInfo have a big enough buffer into which to copy the data, but that will wait until I look at that source.

### CODE REVIEW

File: src/Cache.cpp [hunk 1]
```diff
@@ -115,19 +115,19 @@ tr1::shared_ptr<Frame> Cache::GetSmallestFrame()
...
-	 deque<long int>::iterator itr;
-	 long int smallest_frame = -1;
-	 for(itr = frame_numbers.begin(); itr != frame_numbers.end(); ++itr)
-	 {
-		 if (*itr < smallest_frame || smallest_frame == -1)
-			 smallest_frame = *itr;
-	 }
+	// Loop through frame numbers
+	deque<long int>::iterator itr = frame_numbers.begin();
+	long int smallest_frame = *itr;
```
Comment (komackaj) [inline src/Cache.cpp]: Cannot frame_numbers be empty when calling this? Then itr should not be dereferenced. Maybe check and return NULL if frame_numbers.empty() ?

Comment (RussNelson) [inline src/Cache.cpp]: If that's a problem here, it's a problem in other places in Cache. Oh, and apologizes for including the Cache changes here. I had the wrong branch checked-out.

Comment (komackaj) [inline src/Cache.cpp]: I can't see any problem anywhere else in Cache.cpp - frame_numbers.begin() is always used in for cycle with termination condition `itr != frame_numbers.end()` that will terminate the loop on first check with empty frame_numbers, so it won't dereference in the cycle's body.

Comment (RussNelson) [inline src/Cache.cpp]: Now I'm confused. I was pretty sure I had copied that model from somewhere else in Cache.cpp, but now I can't find it. :-(

**Pull request closed without merging**